python_scripts/libaerie/products/product_parser.py

```diff
@@ -348,13 +582,31 @@ def cast(self, event_hashs: Iterable[dict]) -> None:
       self._fh.write(self.cast_event(event_hash))
       num_r += 1
 
-    logger.info("Encoded %s activities to %s", num_r, self._fh.name)
+    logger.info("Encoded %s activities to %s", num_r, self.filename)
 
-    logger.debug("Closing file: %s", self._fh.name)
-    self._fh.close()
+    #logger.debug("Closing file: %s", self.filename)
```

Comment:
Curious about the commented out code here. Should it be restored or deleted?

Reply:
So, I commented this out to workaround an issue I was having in the tests.  I'm utilizing a file-like buffer to store the contents of the re-encoded file which is compared to a control fixture for correctness.  When the buffer is closed it will destroy the content of the encoding meaning it can't be compared.  I need to add a close() function to the decoder/encoders to close the buffers after cast() finishes running.

Reply:
Nice!  Thanks for the tip.  I overrode the close function in the tests, and enabled the lines again.